keychain_win.cpp

```diff
@@ -170,11 +175,8 @@ namespace keychain {
 void setPassword(const std::string &package, const std::string &service,
                  const std::string &user, const std::string &password,
                  Error &err) {
-    ::SetLastError(0); // clear thread-global error
-
-    auto target_name = makeTargetName(package, service, user);
-    if (!target_name) {
-        updateError(err);
+    auto target_name = makeTargetName(package, service, user, err);
+    if (err) {
```

Comment:
Here we have to know that `target_name` is a `nullptr` exactly `if (err)` and we can safely use it if not. But I think that's okay.

Reply:
Yeah, that's a good point. I'll add a bit of fairy dust to make sure.